src/document.c

```diff
@@ -1955,37 +1961,51 @@ static gchar *write_data_to_disk(const gchar *locale_filename,
 		/* Use POSIX API for unsafe saving (GVFS-unsafe) */
 		/* The error handling is taken from glib-2.26.0 gfileutils.c */
 		errno = 0;
-		fp = g_fopen(locale_filename, "wb");
+		fp = g_fopen(locale_filename, "w+b"); // we can not destroy  symbolic/hard links
```

Comment:
if use "w" in case of appcrash will lost  entire  file.
if use "a" we can not seek to start of file and rewrite it.
if "w+" we can have significant part of file if appcrash, because  truncate+write.